test/e2e/config/docker.yaml

```
@@ -58,25 +58,34 @@ providers:
       new: --metrics-addr=:8080
     files:
     - sourcePath: "../data/shared/v1.0/metadata.yaml"
-  - name: v1.2.11 # supported release in the v1beta1 series; this is used for v1beta1 --> main clusterctl upgrades test only.
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.11/core-components.yaml"
+  - name: v1.2.12 # supported release in the v1beta1 series; this is used for v1beta1 --> main clusterctl upgrades test only.
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.12/core-components.yaml"
     type: "url"
     contract: v1beta1
     replacements:
     - old: --metrics-addr=127.0.0.1:8080
       new: --metrics-addr=:8080
     files:
     - sourcePath: "../data/shared/v1.2/metadata.yaml"
-  - name: v1.3.5 # latest published release in the v1beta1 series; this is used for v1beta1 --> main clusterctl upgrades test only.
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.5/core-components.yaml"
+  - name: v1.3.6 # latest published release in the v1beta1 series; this is used for v1beta1 --> main clusterctl upgrades test only.
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.6/core-components.yaml"
     type: "url"
     contract: v1beta1
     replacements:
       - old: --metrics-addr=127.0.0.1:8080
         new: --metrics-addr=:8080
     files:
       - sourcePath: "../data/shared/v1.3/metadata.yaml"
-  - name: v1.4.99 # next; use manifest from source files
+  - name: v1.4.0 # latest published release in the v1beta1 series; this is used for v1beta1 --> main clusterctl upgrades test only.
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.0/core-components.yaml"
+    type: "url"
+    contract: v1beta1
+    replacements:
+      - old: --metrics-addr=127.0.0.1:8080
+        new: --metrics-addr=:8080
+    files:
+      - sourcePath: "../data/shared/v1.3/metadata.yaml"
```

This looks incorrect to me and the root cause for e2e test failures :

```suggestion
      - sourcePath: "../data/shared/v1.4/metadata.yaml"
```

Also, please make sure to add new `metadata.yaml` file inside the new folder in the following path: `test/e2e/data/shared/v1.4`